File: Samples/azure-storage/Azure.Ruby/azure_storage/models/usage_list_result.rb
# encoding: utf-8

module Petstore
  module Models
    #
    # The List Usages operation response.
    #
    class UsageListResult

      include MsRestAzure

      # @return [Array<Usage>] Gets or sets the list Storage Resource Usages.
      attr_accessor :value


      #
      # Mapper for UsageListResult class as Ruby Hash.
      # This will be used for serialization/deserialization.
      #
      def self.mapper()
        {
          required: false,
          serialized_name: 'UsageListResult',
          type: {
            name: 'Composite',
            class_name: 'UsageListResult',
            model_properties: {
              value: {
                required: false,
                serialized_name: 'value',
                type: {
                  name: 'Sequence',
                  element: {
                      required: false,
                      serialized_name: 'UsageElementType',
                      type: {
                        name: 'Composite',
                        class_name: 'Usage'
                      }
                  }
                }
              }
            }
          }
        }
      end
    end
  end
end

File: Samples/azure-storage/Azure.NodeJS/models/usageListResult.js
/*
 */

'use strict';

var util = require('util');

/**
 * @class
 * Initializes a new instance of the UsageListResult class.
 * @constructor
 * The List Usages operation response.
 *
 */
function UsageListResult() {
}

util.inherits(UsageListResult, Array);

/**
 * Defines the metadata of UsageListResult
 *
 * @returns {object} metadata of UsageListResult
 *
 */
UsageListResult.prototype.mapper = function () {
  return {
    required: false,
    serializedName: 'UsageListResult',
    type: {
      name: 'Composite',
      className: 'UsageListResult',
      modelProperties: {
        value: {
          required: false,
          serializedName: '',
          type: {
            name: 'Sequence',
            element: {
                required: false,
                serializedName: 'UsageElementType',
                type: {
                  name: 'Composite',
                  className: 'Usage'
                }
            }
          }
        }
      }
    }
  };
};

module.exports = UsageListResult;
